### PR TITLE
feat: `Handler.installHandler`

### PR DESCRIPTION
**Pull request by JonathanStarup**
Refactoring
- `ifTrue` -> `ifCondition`
- Thunked the instructions because of recursives defs

Features
- Suspension class (and associated unwind complications)
  - `unwindThunkToTpe` and `unwindSuspensionFreeThunkToTpe` exactly shows where things are expected to be control-pure
  - controlpure exists in
    - Functional java interfaces
    - anonymous java object functions
    - Lazy.force
    - Entrypoints like main and tests
- EffectCall class
- Dup2 instruction
- Corrections to the example code to reflect current design

### CODE REVIEW

**Comment by magnus-madsen** [comment]
CI Errors

**Comment by JonathanStarup** [comment]
@magnus-madsen this PR is less trivial than the others, in regards to where closures are invoked in a control-pure way

**Comment by JonathanStarup** [comment]
image

**Comment by magnus-madsen** [comment]
Not sure I understand the table.

**Comment by magnus-madsen** [comment]
But in any case, adding handlers will have some performance impact.

Hopefully we can regain that later for pure code, e.g. with use of fine-grain CBC, and other tricks.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala [hunk 1]
```diff
@@ -1334,10 +1337,72 @@ object BackendObjType {
...
+        DUP2() ~ GETFIELD(Suspension.PrefixField) ~ // [..., s', s, s', s.prefix]
+        // Make the new frame and push it
+        pushNull() ~ // TODO
+        INVOKEINTERFACE(Frames.PushMethod) ~ // [..., s', s, s', prefix']
+        PUTFIELD(Suspension.PrefixField) ~ // [..., s', s]
+        POP() ~ // [..., s']
+        // Return the suspension up the stack
+        xReturn(Suspension.toTpe)
+      }
+    }
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala:1366]: DSL looks pretty neat 👍

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**